reference-implementation/lib/writable-stream.js

```diff
@@ -932,26 +947,28 @@ function WritableStreamDefaultControllerGetBackpressure(controller) {
 function WritableStreamDefaultControllerError(controller, e) {
   const stream = controller._controlledWritableStream;
 
-  assert(stream._state === 'writable' || stream._state === 'closing');
-
-  const oldState = stream._state;
+  assert(stream._state === 'writable');
 
   stream._state = 'errored';
   stream._storedError = e;
 
-  if (stream._pendingAbortRequest === undefined && stream._writer !== undefined) {
-    let readyPromiseIsPending = false;
-    if (oldState === 'writable' &&
-        WritableStreamDefaultControllerGetBackpressure(stream._writableStreamController) === true) {
-      readyPromiseIsPending = true;
+  // TODO(tyoshino): Given that writer.abort() fails immediately when the stream has been errored, shouldn't
```

Comment:
I think we should try as much as possible to guarantee that the stream is quiescent when abort() returns. This also affects the timing of the return from pipeTo(), since that waits for WritableStreamAbort().